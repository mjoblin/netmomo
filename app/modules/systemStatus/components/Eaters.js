import React from 'react';
import PropTypes from 'prop-types';
import { Table } from 'antd';


export const Eaters = ({ eaters }) => {
    const columns = [
        {
            title: 'Name',
            dataIndex: 'eater_name',
        },
        {
            title: 'Host',
            dataIndex: 'host',
        },
        {
            title: 'Port',
            dataIndex: 'port',
        },
    ];

    return (
        <Table
            dataSource={eaters}
            columns={columns}
            pagination={false}
            rowKey={row => `${row.eater_name}_${row.host}_${row.port}`}
            size="middle"
        />
    );
};


Eaters.propTypes = {
    eaters: PropTypes.array.isRequired,
};

export default Eaters;
