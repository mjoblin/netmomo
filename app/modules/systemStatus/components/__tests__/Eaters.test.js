import React from 'react';
import { shallow } from 'enzyme';
import { Eaters } from '../Eaters';
import testData from './testData.json';

describe('Eaters', () => {
    it('matches snapshot', () => {
        const component = shallow(
            <Eaters
                eaters={testData.dumpling_eaters}
            />
        );

        expect(component).toMatchSnapshot();
    });
});