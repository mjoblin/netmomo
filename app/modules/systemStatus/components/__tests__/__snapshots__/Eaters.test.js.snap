// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Eaters matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Eaters
    eaters={
      Array [
        Object {
          "eater_name": "netmomo",
          "host": "::1",
          "port": 53013,
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "bordered": false,
      "className": "",
      "columns": Array [
        Object {
          "dataIndex": "eater_name",
          "title": "Name",
        },
        Object {
          "dataIndex": "host",
          "title": "Host",
        },
        Object {
          "dataIndex": "port",
          "title": "Port",
        },
      ],
      "dataSource": Array [
        Object {
          "eater_name": "netmomo",
          "host": "::1",
          "port": 53013,
        },
      ],
      "indentSize": 20,
      "loading": false,
      "locale": Object {},
      "pagination": false,
      "prefixCls": "ant-table",
      "rowKey": [Function],
      "rowSelection": null,
      "showHeader": true,
      "size": "middle",
      "useFixedHeader": false,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "bordered": false,
        "className": "",
        "columns": Array [
          Object {
            "dataIndex": "eater_name",
            "title": "Name",
          },
          Object {
            "dataIndex": "host",
            "title": "Host",
          },
          Object {
            "dataIndex": "port",
            "title": "Port",
          },
        ],
        "dataSource": Array [
          Object {
            "eater_name": "netmomo",
            "host": "::1",
            "port": 53013,
          },
        ],
        "indentSize": 20,
        "loading": false,
        "locale": Object {},
        "pagination": false,
        "prefixCls": "ant-table",
        "rowKey": [Function],
        "rowSelection": null,
        "showHeader": true,
        "size": "middle",
        "useFixedHeader": false,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
